pilot/pkg/model/service.go

```diff
@@ -516,12 +516,16 @@ func (ep *IstioEndpoint) IsDiscoverableFromProxy(p *Proxy) bool {
 
 // Metadata returns the endpoint metadata used for telemetry purposes.
 func (ep *IstioEndpoint) Metadata() *EndpointMetadata {
+	labels := map[string]string{}
+	for k, v := range ep.Labels {
+		labels[k] = v
+	}
```

Comment:
Can we deepcopy as needed

Reply:
Did n't understand - what else need to be copied apart from labels?

Reply:
nm, understood what you mean,. Fixed

Reply:
nit: maps.Clone

actually not really just a nit, it will perform better.